### PR TITLE
HSEARCH-4172 Ignore relative ordering of works on different documents in BackendMock

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4172

Should also fix https://hibernate.atlassian.net/browse/HSEARCH-4164

### CODE REVIEW

File: util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/rule/BackendMock.java [hunk 1]
```diff
@@ -421,9 +402,13 @@ public DocumentWorkCallListContext discarded() {
 			log.debugf( "Expecting %d works to be discarded", works.size() );
 			works.stream()
 					.map( work -> new DocumentWorkCall( indexName, DocumentWorkCall.WorkPhase.DISCARD, work ) )
-					.forEach( expectationConsumer );
+					.forEach( this::expect );
 			return this;
 		}
+
+		private void expect(DocumentWorkCall call) {
+			backendBehavior().getDocumentWorkCalls( call.documentKey() ).expectInOrder( call );
```
Comment (fax4ever) [inline util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/rule/BackendMock.java:410]: Does still it make sense to have a method named `expectInOrder`?

**Review by fax4ever — COMMENTED** [review]
It looks nice, I've just a doubt.

File: util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/rule/BackendMock.java (hunk 1 above)
Comment (yrodiere) [reply util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/rule/BackendMock.java:410]: `expectInOrder`/`expectInAnyOrder` are defined in `CallQueue`, which is ised for for other purposes, in particular schema expectations where we need `expectInAnyOrder`.

So yes, it makes sense.

Comment (fax4ever) [reply util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/rule/BackendMock.java:410]: OK +1 to merge it

**Pull request merged**